web/src/HUD.tsx

```diff
@@ -248,16 +239,18 @@ export default class HUD extends Component<HudProps, HudState> {
 
   render() {
     let view = this.state.view
+    let session = this.state.view.uiSession?.status
 
-    let needsNudge = view?.needsAnalyticsNudge ?? false
+    let needsNudge = session?.needsAnalyticsNudge ?? false
     let resources = view?.uiResources ?? []
-    if (!resources?.length || !view?.tiltfileKey) {
+    if (!resources?.length || !session?.tiltfileKey) {
       return <HeroScreen message={"Loading…"} />
     }
 
-    let runningBuild = view?.runningTiltBuild
-    let suggestedVersion = view?.suggestedTiltVersion
-    const versionSettings = view?.versionSettings
+    let tiltfileKey = session?.tiltfileKey || ""
```

Comment:
```suggestion
    let tiltfileKey = session.tiltfileKey
```

Maybe nitty, but I think the `!view?.tiltfileKey` condition in the `if` a few lines up is so that the compiler knows that `view.tiltfileKey` was a `string` in `tiltfileKeyContext.Provider` below, so the `|| ""` seems more confusing than anything. Or maybe something changed so that no longer worked?